cmds/modules/provisiond/reporter.go

```diff
@@ -194,12 +232,21 @@ func (r *Reporter) synchronize(ctx context.Context, reported []Consumption) erro
 
 	// the idea here is that we bring ALL active node contracts from chain.
 	// then compare it with what we have atm (the one we just reported)
-	contracts, err := r.substrate.GetNodeContracts(r.nodeID, substrate.ContractState{IsCreated: true})
+	nodeContracts, err := r.substrate.GetNodeContracts(r.nodeID, substrate.ContractState{IsCreated: true})
 	if err != nil {
 		return err
 	}
 
-	for _, contract := range contracts {
+	for _, contract := range nodeContracts {
+		deployment, err := r.engine.Storage().Get(uint32(contract.TwinID), uint64(contract.ContractID))
+		if err != nil && !errors.Is(err, provision.ErrDeploymentNotExists) {
+			log.Error().
+				Err(err).
+				Msgf("failed to get deployment %d for name contract validation", contract.ContractID)
+		} else if err == nil && r.containsExpiredNameWorkloads(deployment) {
```

Comment:
This code (if it works) will decomission the entire deployment even if one name contract is invalid. May be it need to be re-written so you can decomission a single workload (the invalid name one). the engire already has a method to do so.